src/components/tables/TableHeadingRowConnected.tsx

```diff
@@ -8,29 +8,24 @@ import * as React from 'react';
 import * as _ from 'underscore';
 
 const mapStateToProps = (state: IReactVaporState, ownProps: ITableHeadingRowOwnProps) => {
-  let item: ITableRowState = _.findWhere(state.rows, { id: ownProps.id });
+  const item: ITableRowState = _.findWhere(state.rows, { id: ownProps.id });
 
   return {
-    opened: item && item.opened
+    opened: item && item.opened,
+    selected: item && item.selected,
   };
 };
 
 const mapDispatchToProps = (dispatch: (action: IReduxAction<IReduxActionsPayload>) => void,
   ownProps: ITableHeadingRowOwnProps) => ({
     onClick: () => {
-      if (ownProps.isCollapsible) {
-        dispatch(toggleRow(ownProps.id));
-      }
+      dispatch(toggleRow(ownProps.id, ownProps.isCollapsible));
```

Comment:
The row is always toggled even if not collapsible?

Reply:
I hesitated on this one, but I think it's good to have all the logic happening in one place (the state). Let me know what you think

Reply:
After reading the rest of the PR I think that it's the name of the action that bothers me

Reply:
i've changed it for selectRow and SELECT_ROW and TableRowActions.select, added a check here as well but preserved the logic in the state too in case someone decides to dispatch a selectRow action from somewhere else